Change log for Release 1.8.0 (#2721)
mlflow/mlflow: CHANGELOG.rst

Changelog
=========
1.8.0 (2020-04-16)
------------------
MLflow 1.8.0 includes several major features and improvements:

Features:

> - Csteegz/update azml (#2711, @akshaya-a)

Reviewer: Working on this.

- Case sensitivity and more tests for LIKE and ILIKE (#2708, @mparkhe)
- Change line smoothing on Exponential Moving Average (#2620, @Valentyn1997)
- Add support for spacy flavor (#2242, @arocketman)
- [UI] Separate selection checkbox into its own column (#2559, @harupy)
- highlight changes in CompareRunView (#2565, @gabrielbretschner)
- Add run comparison in model registry (#2537, @juntai-zheng)
- RFC: Allow string experiment IDs in the UI (#2568, @jonas)
- Passing args to docker run CLI (#2608, @ksanjeevan)
- - added windows support for pushing images to aws ecr (#2500, @AndreyBulezyuk)

Bug fixes and documentation updates:

- Fix CLI summary message in mlflow azureml build_image CLI (#2712, @dbczumar)
- update flower example (#2647, @tfurmston)

> - Let curl follow redirects (fix #2691) (#2699, @HiromuHota)

Reviewer 2: For these bug fixes / doc updates, it'd be good to clearly state what use case is benefitted by the bugfix. For example, in this case, from the description in #2691 it looks like we fixed a bug in `mlflow models build-docker`, so we could say something like `"Fixed pip not found error while packaging models via mlflow models build-docker (#2699, @HiromuHota)"`

See https://github.com/mlflow/mlflow/blob/master/CHANGELOG.rst#170-2020-03-02 for more examples on wording from previous changelogs

Reviewer 2: Since there are a number of these in this release (& users likely are more directly-impacted by bugfixes than documentation updates), it's probably worth it to split them out into separate `Bug fixes` & `Documentation updates` sections, with bug fixes listed first.

Changelog
=========
1.8.0 (2020-04-16)
------------------
MLflow 1.8.0 includes several major features and improvements:

Features:

> - Added new APIs for Azure Deployment (#2375 @csteegz, #2711, @akshaya-a)

Reviewer 2: ```suggestion
- Added new `mlflow.azureml.deploy` API for deploying MLflow models to AzureML (#2375 @csteegz, #2711, @akshaya-a)
```

Reviewer: Done

> - Added support for case sensitive LIKE and ILIKE operator in SQLite (#2708, @mparkhe)

Reviewer 2: ```suggestion
- You can now perform case-sensitive LIKE queries (e.g. `'params.framework LIKE '%sklearn%'`) with the SearchRuns API & UI when running against a SQLite backend (#2708, @mparkhe)
```

Reviewer: Done.

Reviewer 3: This should be credit as `(#2217, @t-henri; #2708, @mparkhe)`

Also fix the message like

```
- Added support for case-sensitive LIKE and case-insensitive ILIKE queries (e.g. ``'params.framework LIKE '%sklearn%'``) with the SearchRuns API & UI when running against a SQLite backend (#2217, @t-henri; #2708, @mparkhe)
```

Reviewer: Done

> - Added line smoothing on Exponential Moving Average (#2620, @Valentyn1997)

Reviewer 2: ```suggestion
- Improved line smoothing in MLflow metrics UI using exponential moving averages (#2620, @Valentyn1997)
```

Reviewer: Done.

> - Adding support for spacy flavor (#2242, @arocketman)

Reviewer 2: ```suggestion
- Added `mlflow.spacy` module with support for logging and loading spaCy models (#2242, @arocketman)
```

Reviewer: Done

> - Separating selection checkbox in mlflow UI to have it's own column (#2559, @harupy)

Reviewer 2: I would actually move this one to bug fixes, and say it "Fixes a bug where hiding the start time column in the UI would also hide run selection checkboxes"

Reviewer: Done

> - Highlighting different parameter when comparing runs (#2565, @gabrielbretschner)

Reviewer 2: ```suggestion
- Highlight parameter values that differ across runs in run comparison UI (#2565, @gabrielbretschner)
```

Reviewer: Done

> - Add run comparison in model registry (#2537, @juntai-zheng)

Reviewer 2: ```suggestion
- Adds ability to compare source runs associated with model versions from the models UI  (#2537, @juntai-zheng)
```

Reviewer: Done

> - Adding UI support for non-numerical experiment ID's (#2568, @jonas)

Reviewer 2: ```suggestion
- Add support for viewing experiments with non-numerical experiment in the UI (#2568, @jonas)
```

Reviewer 4: `non-numerical IDs`?

Reviewer: Thank you for pointing this out, I have fixed it.

- Docker run now accepts arguments in CLI (#2608, @ksanjeevan)
- Adding windows support for pushing images to aws ecr (#2500, @AndreyBulezyuk)

Bug fixes:

> - Fixed CLI summary message in mlflow azureml build_image CLI (#2712, @dbczumar)

Reviewer 2: ```suggestion
- Fixed CLI summary message in `mlflow azureml build_image` CLI (#2712, @dbczumar)
```

BTW, you can verify the changelog formatting by viewing the file at https://github.com/mlflow-automation/mlflow/blob/2020-04-16-19-30-04-063481/CHANGELOG.rst

Reviewer: Ohh yea, there is also a preview option here.

> - Updated flower_classifier/score_images_rest.py with multiple bug fixes (#2647, @tfurmston)

Reviewer 2: ```suggestion
- Updated examples/flower_classifier/score_images_rest.py with multiple bug fixes (#2647, @tfurmston)
```

Reviewer: Done.

> - Fixed pip not found error while packaging models via mlflow models build-docker (#2699, @HiromuHota)

Reviewer 2: ```suggestion
- Fixed pip not found error while packaging models via `mlflow models build-docker` (#2699, @HiromuHota)
```

Reviewer: Done

> - Fixed bug causing erroneous deletion of TensorBoard logging directory (#2670, @dbczumar)

Reviewer 2: ```suggestion
- Fixed bug in `mlflow.tensorflow.autolog` causing erroneous deletion of TensorBoard logging directory (#2670, @dbczumar)
```

Reviewer: Done

- Fixed a bug that truncated the description of the mlflow gc subcommand in mlflow --help (#2679, @dbczumar)

> - Fixed bug where Travis builds and MLFlow Docker packaging was failing due to incorrect Miniconda download URL (#2685, @michaeltinsley)

Reviewer 2: ```suggestion
- Fixed bug where ``mlflow models build-docker`` was failing due to incorrect Miniconda download URL (#2685, @michaeltinsley)
```

Don't need to mention the Travis fix here as it's not relevant to users

Reviewer: Done.

- Fixed a bug where boto3 was not using $MLFLOW_S3_ENDPOINT_URL even though it was specified (#2629, @poppash)
- Fixed a bug where Sqlite in-memory was not working as a backend store by modifying DB upgrade logic (#2667, @dbczumar)
- Fixed a bug to allow numerical parameters with values >= 1000 in R mlflow::mlflow_run() (#2665, @lorenzwalthert)
- Fixed a bug where AWS creds was not found in the Windows platform due path differences (#2634, @AndreyBulezyuk)
- Fixed a bug to add pip when necessary in _mlflow_conda_env (#2646, @tfurmston)
- Fixed error code to be more meaningful if input to model version is incorrect (#2625, @andychow-db)

> - Fixed stage name issue in .travis.yml (#2632, @keigohtr)

Reviewer 2: ```suggestion
```
This is a great contribution, but would mention it in the "Small bugfixes and doc updates", as it's not user-visible

Reviewer: Done.

> - Fixed multiple bugs in model registry (#2638, @aarondav)

Reviewer 2: Could we ask @aarondav for a more detailed description of what bugs were fixed here?
